### PR TITLE
Player tokens

### LINKED ISSUE #67: Return player IDs to frontend
Due to the way [React handles keys](https://reactjs.org/docs/lists-and-keys.html) we need to return a unique attribute for each player. Currently each `nickname` is unique and we keep a players MongoDb `_id` field secret as this is used for auth.

However, if a player (Dwight) leaves the lobby then re-joins with the same nickname the following appears for other players:

182884847_3907560555986676_7339567978499685635_n (1)

As a result, we should return each players `_id` to the frontend and instead define a new `token` UUID field for each player that is kept secret and used for auth.

### PR DESCRIPTION
**Pull request by taitfuller**
Closes #67 

- Adds `token` UUID to Player schema
- Refactors POST /player/create to return a payload of `{ playerId: XXX, token: YYY }`, updating frontend API calls to consume this
- Replaces `validatePlayerId` service with `authenticatePlayer`
- Returns each players `id` on `players:initial`

### CODE REVIEW

File: backend/src/models/__tests__/game.model.test.ts [hunk 1]
```diff
@@ -146,6 +147,25 @@ describe("Game Model", () => {
...
+  it("throws a ValidationError if a player is added without a nickname", async () => {
+    gameData.players = [
+      { punchlines: ["To get to the other side"] },
+      { punchlines: ["To go to KFC"] },
+    ];
+
+    const game = new GameModel(gameData);
+
+    try {
+      await game.save();
```
Comment (RawiriHohepa) [inline backend/src/models/__tests__/game.model.test.ts:164]: should put a fail() in here to make sure it throws an error

File: backend/src/services/__tests__/player.service.test.ts [hunk 2]
```diff
@@ -34,11 +35,16 @@ describe("createPlayer Service", () => {
...
 
-    const playerId = await createPlayer(gameCode, "Bob");
+    const { playerId, token } = await createPlayer(gameCode, "Bob");
 
     const game = await getGame(gameCode);
 
-    expect(game.players.id(playerId)).not.toBeNull();
+    const player = game.players.id(playerId);
+    expect(player).toBeDefined();
+    if (player === null) return;
```
Comment (RawiriHohepa) [inline backend/src/services/__tests__/player.service.test.ts:44]: doesn't return pass the test? Is this a valid state for player, or should it run fail()?

**Review by RawiriHohepa — APPROVED** [review]
Looks good for the most part, just a couple of places in tests where you could add a fail() statement (or jest's equivalent)

File: backend/src/models/__tests__/game.model.test.ts (hunk 1 above)
Comment (taitfuller) [reply backend/src/models/__tests__/game.model.test.ts:164]: Yep good catch!

File: backend/src/services/__tests__/player.service.test.ts (hunk 2 above)
Comment (taitfuller) [reply backend/src/services/__tests__/player.service.test.ts]: Yep good catch!

**Review by RawiriHohepa — APPROVED** [review]
Looks good!

**Pull request merged**